### PR TITLE
Use namespace in PyLinter and tests

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

Almost there...

### CODE REVIEW

File: pylint/lint/pylinter.py [hunk 1]
```diff
@@ -839,6 +839,12 @@ def _parse_error_mode(self) -> None:
 
         self.disable_noerror_messages()
         self.disable("miscellaneous")
+        self._arg_parser.parse_args(
```
Comment (Pierre-Sassoulas) [inline pylint/lint/pylinter.py:842]: Hmm that's strange, why do we need that ? It's overriding actual values, right ?

Comment (DanielNoord) [reply pylint/lint/pylinter.py:842]: I haven't implemented a `set_option` that alters `namespace` yet. The previous one only does `config`. However, the current `set_option` is defined on `BaseChecker`. That's strange as it should be on `OptionsManager`.

I'm planning to implement `set_option` on `ArgumentsManager` but I need #6187 for it. I thought in the mean time this was good enough as the TODO makes sure I'll revisit and I'll probably do a search for `_arg_parser.parse_args` at the end anyway.
This keeps the ball rolling so to speak 😄

Comment (Pierre-Sassoulas) [reply pylint/lint/pylinter.py:842]: But to be clear it means users won't be able to change the score or report value themself ?

Comment (DanielNoord) [reply pylint/lint/pylinter.py:842]: No, that's what `errors-only` mode does. See the 3 lines just under the fix I added here. They do essentially the same as this `parse_args` only they affect `linter.config` and this changes `linter.namespace`.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**